linkerd/app/gateway/src/config.rs

```diff
@@ -48,30 +48,22 @@ impl Config {
                 profiles,
                 Allow(self.allow_discovery),
             ))
-            .check_new_service::<inbound::Target, http::Request<http::BoxBody>>()
-            .instrument(|_: &inbound::Target| debug_span!("gateway"))
+            .instrument(
+                |g: &inbound::HttpGatewayTarget| debug_span!("gateway", target=%g.target, v=%g.version),
```

Comment:
nit: in general we like to put whitespace around the `=` i think:
```suggestion
                |g: &inbound::HttpGatewayTarget| debug_span!("gateway", target = %g.target, v = %g.version),
```